api/v1alpha1/profilecatalogsource_types.go

```diff
@@ -19,13 +19,30 @@ package v1alpha1
 import (
 	"strings"
 
+	"github.com/fluxcd/pkg/apis/meta"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // ProfileCatalogSourceSpec defines the desired state of ProfileCatalogSource
 type ProfileCatalogSourceSpec struct {
 	// Profiles is the list of profiles exposed by the catalog
+	// +optional
 	Profiles []ProfileCatalogEntry `json:"profiles,omitempty"`
+	// Repositories is the list of repositories to scan for profiles
+	// +optional
+	Repos []Repository `json:"repositories,omitempty"`
+}
+
+type Repository struct {
+	// URL is the URL of the repository
+	URL string `json:"url,omitempty"`
+	// The secret name containing the Git credentials.
+	// For HTTPS repositories the secret must contain username and password
```

Comment:
What about tokens? Should we include that or just say, in case of tokens the password is the token and the username is not required?

Reply:
don't think a token is supported https://fluxcd.io/docs/components/source/gitrepositories/#https-authentication

Reply:
In our docs we will just link to the flux docs, since we are mirroring them in authentication

Reply:
Ah okay